Client/Frontend/Browser/FrequencyQuery.swift

```diff
@@ -13,41 +13,70 @@ class FrequencyQuery {
 
   private let historyAPI: BraveHistoryAPI
   private let bookmarkManager: BookmarkManager
-  private let queue = DispatchQueue(label: "frequency-query-queue")
-  private var cancellable: DispatchWorkItem?
+  private let tabManager: TabManager
 
-  init(historyAPI: BraveHistoryAPI, bookmarkManager: BookmarkManager) {
+  private let frequencyQueue = DispatchQueue(label: "frequency-query-queue")
+  private var task: DispatchWorkItem?
+    
+  init(historyAPI: BraveHistoryAPI, bookmarkManager: BookmarkManager, tabManager: TabManager) {
     self.historyAPI = historyAPI
     self.bookmarkManager = bookmarkManager
+    self.tabManager = tabManager
   }
 
-  public func sitesByFrequency(containing query: String, completion: @escaping (Set<Site>) -> Void) {
-    historyAPI.byFrequency(query: query) { [weak self] historyList in
-      guard let self = self else {
-        completion(Set<Site>())
-        return
-      }
+  deinit {
+    task?.cancel()
+  }
 
-      let historySites =
-        historyList
-        .map { Site(url: $0.url.absoluteString, title: $0.title ?? "") }
+  public func sitesByFrequency(containing query: String, completion: @escaping (Set<Site>) -> Void) {
+    task = DispatchWorkItem {
+      // brave-core fetch can be slow over 200ms per call,
+      // a cancellable serial queue is used for it.
+      DispatchQueue.main.async {
```

Comment:
assuming you are not on 13.3/swift 5.6 cuz the this should give a new warning of not having an explicit `self`

Reply:
No warning here with 13.3/swift 5.6